react-app/src/account/Account.test.js

```diff
@@ -0,0 +1,52 @@
+import Account, { ErrorAuthenticationFail } from './Account.js';
+import naclUtils from 'tweetnacl-util';
+
+let account;
+
+const mockUserData = {
+  username:           '$/()RUFJ)wuwe84349',
+  password:           'dnefu384(/·$873hf7g',
+  expectedPublicKey:  '6FUygceUHlKzGu/5ir4FNkmAtUF07uNeAAdp9+5hcy4=',
+  expectedSecretKey:  'm4jBZuDiCgcCz94VIWHmeVC9IsXOIavniA2pqeq5Gg0='
+};
+
+test('match between generated key pairs with expected ones', () => {
+  expect.assertions(5);
+  account = new Account(mockUserData.username, mockUserData.password);
+
+  expect(account._username).toBe(mockUserData.username);
+
+  const systemTzOffsetInMinutes = -(new Date()).getTimezoneOffset();
```

Comment:
Using a variable value is not a good idea because the test may fail for one value but pass with another with no reliable way of reproducing it.
This (or just playing with random values) is useful to come up with test cases, but the real test cases should have a fixed value. That way the test either fails or pass, but always the same, no matter the day or where you are at.

There must not be a conditional as for `systemTz`. We should have a test that says something like "given that the time is `3`, I expect to get `3-whatever`".

Reply:
I wrote this piece of code most as a guard than a test itself. The problem I found here is that the offset and the timezone are computed using the user's timezone and if I set this value for a test, it will only success if the user's timezone is the same as the one fixed for the test.

Reply:
Done